csharp/src/Apache.Arrow/Arrays/StringArray.cs

```diff
@@ -26,6 +26,8 @@ public class StringArray: BinaryArray, IReadOnlyList<string>
     {
         public static readonly Encoding DefaultEncoding = Encoding.UTF8;
 
+        private Dictionary<Encoding, List<string>> materializedStringStore = new Dictionary<Encoding, List<string>>();
```

Comment:
Can the dictionary be allocated lazily so as not to pay the allocation cost when the feature is not in use?

Can the materialized list be a `string[]` instead of a `List<string>` to save an allocation?

Reply:
sure, next iteration.